go-controller/pkg/node/gateway_udn.go

```diff
@@ -0,0 +1,154 @@
+func (udng *UserDefinedNetworkGateway) AddNetwork() error {
+	return udng.addUDNManagementPort()
+}
+
+// DelNetwork will be responsible to remove all plumbings
+// used by this UDN on the gateway side
+func (udng *UserDefinedNetworkGateway) DelNetwork() error {
+	return udng.deleteUDNManagementPort()
+}
+
+// addUDNManagementPort does the following:
+// STEP1: creates the OVS interface on br-int for the UDN's management port interface
+// STEP2: It saves the MAC address generated on the 1st go as an option on the OVS interface
+// so that it persists on reboots
+// STEP3: creates the management port netdevice on the host
+// STEP4: adds the management port IP .2 to the netdevice
+func (udng *UserDefinedNetworkGateway) addUDNManagementPort() error {
+	var err error
+	interfaceName := util.GetNetworkScopedK8sMgmtHostIntfName(uint(udng.networkID))
+	var networkLocalSubnets []*net.IPNet
+	if udng.TopologyType() == types.Layer3Topology {
+		networkLocalSubnets, err = util.ParseNodeHostSubnetAnnotation(udng.node, udng.GetNetworkName())
+		if err != nil {
+			return fmt.Errorf("waiting for node %s to start, no annotation found on node for network %s: %v",
+				udng.node.Name, udng.GetNetworkName(), err)
+		}
+	} else if udng.TopologyType() == types.Layer2Topology {
+		globalFlatL2Networks := udng.Subnets()
+		for _, globalFlatL2Network := range globalFlatL2Networks {
+			networkLocalSubnets = append(networkLocalSubnets, globalFlatL2Network.CIDR)
+		}
+	}
+	networkMTU := udng.NetInfo.MTU()
+	if networkMTU == 0 {
+		networkMTU = config.Default.MTU
+	}
+
+	// STEP1
+	stdout, stderr, err := util.RunOVSVsctl(
+		// delete if any port exists just to clean things up
+		"--", "--if-exists", "del-port", "br-int", interfaceName,
+		"--", "--may-exist", "add-port", "br-int", interfaceName,
+		"--", "set", "interface", interfaceName,
+		"type=internal", "mtu_request="+fmt.Sprintf("%d", networkMTU),
+		"external-ids:iface-id="+udng.GetNetworkScopedK8sMgmtIntfName(udng.node.Name),
+	)
+	if err != nil {
+		return fmt.Errorf("failed to add port to br-int for network %s, stdout: %q, stderr: %q, error: %v",
+			udng.GetNetworkName(), stdout, stderr, err)
+	}
+	klog.V(3).Infof("Added OVS management port interface %s for network %s", interfaceName, udng.GetNetworkName())
+
+	// STEP2
+	macAddress, err := util.GetOVSPortMACAddress(interfaceName)
+	if err != nil {
+		return fmt.Errorf("failed to get management port MAC address for network %s: %v", udng.GetNetworkName(), err)
+	}
+	// persist the MAC address so that upon node reboot we get back the same mac address.
+	_, stderr, err = util.RunOVSVsctl("set", "interface", interfaceName,
+		fmt.Sprintf("mac=%s", strings.ReplaceAll(macAddress.String(), ":", "\\:")))
+	if err != nil {
+		return fmt.Errorf("failed to persist MAC address %q for %q while plumbing network %s: stderr:%s (%v)",
+			macAddress.String(), interfaceName, udng.GetNetworkName(), stderr, err)
+	}
+
+	// STEP3
+	mplink, err := util.LinkSetUp(interfaceName)
```

Comment:
STEP3 doesn't create a device on the host as per the comments. The device is created when you add-port. You are simply just bringing the link state up here. Probably want to reword the comments.

Reply:
yup nice catch! changing